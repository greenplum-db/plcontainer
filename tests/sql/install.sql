../../management/sql/plcontainer_install.sql