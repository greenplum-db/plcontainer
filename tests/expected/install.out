-- Creating PL/Container trusted language
CREATE OR REPLACE FUNCTION plcontainer_call_handler()
RETURNS LANGUAGE_HANDLER
AS '$libdir/plcontainer' LANGUAGE C;
CREATE TRUSTED LANGUAGE plcontainer HANDLER plcontainer_call_handler;
-- Defining container configuration management functions
CREATE OR REPLACE FUNCTION plcontainer_show_local_config() RETURNS text
AS '$libdir/plcontainer', 'show_plcontainer_config'
LANGUAGE C VOLATILE;
CREATE OR REPLACE FUNCTION plcontainer_refresh_local_config(verbose bool) RETURNS text
AS '$libdir/plcontainer', 'refresh_plcontainer_config'
LANGUAGE C VOLATILE;
CREATE OR REPLACE VIEW plcontainer_show_config as
    select gp_segment_id, plcontainer_show_local_config()
        from (
            select gp_segment_id
                from gp_dist_random('pg_namespace')
                group by 1
            ) as segments
    union all
    select -1, plcontainer_show_local_config();
CREATE OR REPLACE VIEW plcontainer_refresh_config as
    select gp_segment_id, plcontainer_refresh_local_config(false)
        from (
            select gp_segment_id
                from gp_dist_random('pg_namespace')
                group by 1
            ) as segments
    union all
    select -1, plcontainer_refresh_local_config(false);
